### PR TITLE
Build ruby wrappers with CMake

### PR DESCRIPTION
**Pull request by isuruf**
This is an initial version. I don't like `../../` paths, maybe we can figure something out later

### CODE REVIEW

**Comment by certik** [comment]
```
CMake Error at src/ruby/CMakeLists.txt:1 (add_subdirectoy):
  Unknown CMake command "add_subdirectoy".
```

**Comment by certik** [comment]
``` diff
diff --git a/src/ruby/CMakeLists.txt b/src/ruby/CMakeLists.txt
index 97e7159..c810936 100644
--- a/src/ruby/CMakeLists.txt
+++ b/src/ruby/CMakeLists.txt
@@ -1 +1 @@
-add_subdirectoy(lib/symengine)
+add_subdirectory(lib/symengine)
```

**Comment by certik** [comment]
With my patch above it works!

File: src/ruby/ext/symengine/symengine.c [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#include "extconf.h"
```
Comment (abinashmeher999) [inline src/ruby/ext/symengine/symengine.c:1]: @isuruf Should we remove this? This contains macros that might be useful later.

Comment (isuruf) [inline src/ruby/ext/symengine/symengine.c:1]: When is this file generated?
What does it contain?

Comment (abinashmeher999) [inline src/ruby/ext/symengine/symengine.c:1]: It's generated right after executing extconf.rb and contains the outputs of have_\* functions as HAVE_\* macros. The contents are given at the end of every mkmf.log file. like

``` cpp
extconf.h is:
/* begin */
1: #ifndef EXTCONF_H
2: #define EXTCONF_H
3: #define HAVE_CWRAPPER_H 1
4: #endif
/* end */
```

Comment (isuruf) [inline src/ruby/ext/symengine/symengine.c:1]: Ah then, we don't need it.

Comment (abinashmeher999) [inline src/ruby/ext/symengine/symengine.c:1]: +1 from my side then. Anything else do you need to add before I merge this?

Comment (isuruf) [inline src/ruby/ext/symengine/symengine.c:1]: Nope. We'll figure out what is needed in the main PR when reviewing

**Pull request merged**